File: src/models/Banners.ts
interface Buttons {
  id: number,
  title: string,
  add_class: string,
  link: string
}

class Banners{
    title:string|null;
    subtitle:string|null;
    image:string|null;
    created_at:Date|null;
    updated_at:Date|null;
    deleted_at:Date|null;
    id:number|null;
    menu_id:number|null;
    buttons: Buttons[];
    label:string|null;
    blog_form:boolean|null;

    constructor(
        title:string|null,
        subtitle:string|null,
        image:string|null,
        created_at:Date|null, 
        updated_at:Date|null, 
        deleted_at:Date|null, id:number|null=null,
        menu_id:number|null=null,
        buttons: Buttons[]=[],
        label: string|null=null,
        blog_form: boolean|null = null,
    ){
        this.title = title;
        this.subtitle = subtitle;
        this.image = image;
        this.created_at = created_at;
        this.updated_at = updated_at;
        this.deleted_at = deleted_at;
        this.id = id;
        this.menu_id = menu_id;
        this.buttons = buttons;
        this.label = label;
        this.blog_form = blog_form;
    }
};
export default Banners;

File: src/main/java/pe/tarjetaestilos/models/Banners.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package pe.tarjetaestilos.models;

/**
 *
 * @author julito
 */
import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;
import org.hibernate.annotations.Where;
import java.time.ZonedDateTime;
import java.util.ArrayList;
import java.util.List;


@Entity
@Table(name="banners")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Where(clause = "deleted_at IS NULL")
public class Banners {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(columnDefinition = "TEXT")
    private String title;
    
    @Column(columnDefinition = "TEXT")
    private String subtitle;
    
    @Column(columnDefinition = "TEXT")
    private String image;
        
    @Column(columnDefinition = "TEXT")
    private String label;
    
    @ManyToOne()
    @JoinColumn(name = "menu_id", nullable = false)
    @ToString.Exclude
    @JsonBackReference
    private Menu menu;
    
    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private ZonedDateTime createdAt;
    
    @UpdateTimestamp
    @Column(name = "updated_at")
    private ZonedDateTime updatedAt;
    
    @Column(name = "deleted_at")
    private ZonedDateTime deletedAt;
    
    @Column(name = "blog_form", nullable = true)
    private Boolean blog_form;
    
    
    @OneToMany(mappedBy = "banners", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Button> buttons = new ArrayList<>();
    
    
}
